skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldFileService.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.skaffold
+
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.search.FileTypeIndex
+import com.intellij.psi.search.GlobalSearchScope
+import org.jetbrains.yaml.YAMLFileType
+import java.io.ByteArrayInputStream
+import java.io.InputStream
+import java.util.Scanner
+import java.util.regex.Pattern
+
+// see https://github.com/GoogleContainerTools/skaffold/blob/master/examples/annotated-skaffold.yaml
+private const val SKAFFOLD_API_HEADER_REGEX = """\s*apiVersion:\s*skaffold/v"""
+private val SKAFFOLD_API_HEADER_PATTERN: Pattern by lazy {
+    Pattern.compile(SKAFFOLD_API_HEADER_REGEX)
+}
+
+/** Obtains current active implementation of [SkaffoldFileService] */
+fun getSkaffoldFileService() = ServiceManager.getService(SkaffoldFileService::class.java)!!
```

Comment:
Was looking at how Kotlin does companion objects. I think something like this _may_ work:

```
class SkaffoldFileService {

  companion object {
    val instance: SkaffoldFileService
      get() = ServiceManager.getService(SkaffoldFileService::class.java)!!
  }  

  fun someOtherFun() ...

}
```
then use it 
```
SkaffoldFileService.instance.someOtherFun()
```

Makes the calling code a bit cleaner. 

wdyt?

Note: I'm not really sure what the difference is between:
```
SkaffoldFileService.instance.someOtherFun()
```
and 
```
SkaffoldFileService.Companion.instance.someOtherFun()
```
as they both seem to compile and do the same thing

Reply:
to answer my question at the end: I think the `.Companion` syntax is for calling Kotlin code from Java. In Kotlin, both work.

Reply:
Looks good, I agree this look more familiar for us and it seems to be a completely normal practice to define single instance of the object. Changed to use companion object and instance field. Thanks!